capsicum-test-main.cc

```diff
@@ -27,6 +30,40 @@ class SetupEnvironment : public ::testing::Environment
     }
     std::cerr << tmpdir << std::endl;
   }
+  void CheckCapsicumSupport() {
+    // For versions of googletest that lack GTEST_SKIP.
+#ifndef GTEST_SKIP
+#define GTEST_SKIP GTEST_FAIL
+#define GTEST_SKIP_defined
+#endif
+
+#ifdef __FreeBSD__
+    size_t trap_enotcap_enabled_len;
+    int rc;
+    bool trap_enotcap_enabled;
```

Comment:
Is this really a `bool` not an `int`?  I'd have guessed that `sysctlbyname(3)` would expect a C89 type...

Reply:
It does, actually. From `sys/sysctl.h`:
```
 376 /* Oid for a bool.  If ptr is NULL, val is returned. */
 377 #define SYSCTL_NULL_BOOL_PTR ((bool *)NULL)
 378 #define SYSCTL_BOOL(parent, nbr, name, access, ptr, val, descr) \
 379         SYSCTL_OID(parent, nbr, name,                           \
 380             CTLTYPE_U8 | CTLFLAG_MPSAFE | (access),             \
 381             ptr, val, sysctl_handle_bool, "CU", descr);         \
 382         CTASSERT(((access) & CTLTYPE) == 0 &&                   \
 383             sizeof(bool) == sizeof(*(ptr)))
```

Here's the commit that added it:
```
commit d09154a9e830f0d400ec0447f6532a6aaeeab001
Author: hselasky <hselasky@FreeBSD.org>
Date:   Thu May 26 08:41:55 2016 +0000

    Add support for boolean sysctl's.
    
    Because the size of bool can be implementation defined, make a bool
    sysctl handler which handle bools. Userspace sees the bools like
    unsigned 8-bit integers. Values are filtered to either 1 or 0 upon
    read and write, similar to what a compiler would do.
    
    Requested by:   kmacy @
    Sponsored by:   Mellanox Technologies

Notes:
    svn path=/head/; revision=300718
```

Reply:
Cool, thanks for checking.